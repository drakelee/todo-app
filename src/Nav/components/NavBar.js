import React, {Component} from 'react'
import AppBar from '@material-ui/core/es/AppBar/AppBar'
import Toolbar from '@material-ui/core/es/Toolbar/Toolbar'
import Typography from '@material-ui/core/es/Typography/Typography'
import IconButton from '@material-ui/core/es/IconButton/IconButton'
import MenuIcon from '@material-ui/icons/Menu'

class NavBar extends Component {
    render() {
        const {title, classes} = this.props
        return (
            <div className={classes.root}>
                <AppBar position='static'>
                    <Toolbar>
                        <IconButton className={classes.menuButton} color='inherit'>
                            <MenuIcon/>
                        </IconButton>
                        <Typography variant='title' color='inherit' className={classes.flex}>
                            {title}
                        </Typography>
                    </Toolbar>
                </AppBar>
            </div>
        )
    }
}

export default NavBar
