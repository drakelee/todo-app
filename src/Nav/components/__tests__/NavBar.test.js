import React from 'react'
import {createShallow} from '@material-ui/core/test-utils'
import NavBar from '../NavBar'
import renderer from 'react-test-renderer'

const shallow = createShallow()
const setup = overrideProps => {
    const classes = {
        menuButton: '',
        root: {},
        flex: ''
    }
    const wrapper = shallow(
        <NavBar
            classes={classes}
            {...overrideProps}
        />
    )
}

describe('<NavBar/>', () => {
    it('renders correctly', () => {
        const classes = {
            menuButton: '',
            root: {},
            flex: ''
        }
        const tree = renderer.create(
            <NavBar
                classes={classes}
            />
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
