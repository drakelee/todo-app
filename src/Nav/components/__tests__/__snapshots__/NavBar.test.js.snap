// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> renders correctly 1`] = `
<div
  className={Object {}}
>
  <header
    className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7"
  >
    <div
      className="MuiToolbar-root-36 MuiToolbar-regular-38 MuiToolbar-gutters-37"
    >
      <button
        className="MuiButtonBase-root-46 MuiIconButton-root-40 MuiIconButton-colorInherit-41"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-45"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-49"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-56"
        />
      </button>
      <h2
        className="MuiTypography-root-63 MuiTypography-title-69 MuiTypography-colorInherit-82"
      />
    </div>
  </header>
</div>
`;
